gc/CMakeLists.txt

```diff
@@ -323,7 +323,7 @@ if(OMR_ENHANCED_WARNINGS)
 endif()
 
 if(CMAKE_CXX_COMPILER_ID MATCHES "GNU|Clang|AppleClang")
```

Comment:
We can delete this whole block of cmake, the xlc option `-qalias=noansi` is set for all files in [xlc.cmake ](https://github.com/eclipse/omr/blob/589f93bb883463c3c3444d58710826cb7c81d15e/cmake/modules/platform/toolcfg/xlc.cmake#L28)